### PR TITLE
feat: 🎸 NFT watch

### PR DESCRIPTION
**Pull request by atoulme**
Add the ability to select contracts to watch for NFT transfers.

### CODE REVIEW

File: docs/configuration.md [hunk 1]
```diff
@@ -270,6 +272,28 @@ Balance watcher is a component tracking transfers of a token and reporting balan
...
+| `-`  | map<string,[`NFTWatcher`](#NFTWatcher)> | Mapping of name => nft watcher.<br><br>See NFTWatcherConfigSchema |
+
+### NFTWatcher
+
+NFT watcher is a component tracking transfers of NFTs and reporting their metadata.
+
+| Name                 | Type                        | Description                                                                                       |
+| -------------------- | --------------------------- | ------------------------------------------------------------------------------------------------- |
+| `contractAddress`    | `string`                    | The address of the contract to watch.                                                             |
+| `startAt`            | [`StartBlock`](#StartBlock) | If no checkpoint exists (yet), this specifies which block should be chosen as the starting point. |
```
Comment (ccordi) [inline docs/configuration.md:290]: endAt could be good to add in the future for testing

File: src/nftwatcher.ts [hunk 2]
```diff
@@ -0,0 +1,385 @@
...
+                debug('%s uses an unsupported protocol', tokenURI);
+                return;
+            }
+        } catch (_) {
+            debug('%s is not a supported url', tokenURI);
+            return;
+        }
+
+        try {
+            const metadataResponse = await fetch(tokenURI);
```
Comment (ccordi) [inline src/nftwatcher.ts:364]: optionally, size can be passed into fetch as an option, which would throw a FetchError on oversized responses.

**Review by ccordi — APPROVED** [review]
Some optional comments but overall looks good for an opt-in feature.

File: docs/configuration.md (hunk 1 above)
Comment (atoulme) [reply docs/configuration.md:291]: Added

File: src/nftwatcher.ts (hunk 2 above)
Comment (atoulme) [reply src/nftwatcher.ts]: Thanks, that's neat- moved to use that

**Pull request merged**